apps/server/test/lexical/server/code_intelligence/entity_test.exs

```diff
@@ -299,30 +299,249 @@ defmodule Lexical.Server.CodeIntelligence.DefinitionTest do
     test "find the definition when calling a erlang module", %{project: project} do
       subject_module = ~q[
         :erlang.binary_to_atom|("1")
-      ]
+        ]
       {:ok, uri, definition_line} = definition(project, subject_module)
 
       assert uri =~ "/src/erlang.erl"
       assert definition_line =~ ~S[«binary_to_atom»(Binary)]
     end
   end
 
-  defp caller_position(subject_module) do
-    {line, character} = cursor_position(subject_module)
-    Position.new(line, character)
+  describe "module resolve/2" do
+    test "succeeds with trailing period", %{project: project} do
+      code = ~q[
+        Some.Modul|e.
+      ]
+
+      assert {:ok, {:module, Some.Module}, range} = resolve(project, code)
+      assert range.start.line == 1
+      assert range.start.character == 1
+      assert range.end.line == 1
+      assert range.end.character == 12
+    end
+
+    test "succeeds immediately following the module", %{project: project} do
+      code = ~q[
+        Beyond.The.End|
+      ]
+
+      assert {:ok, {:module, Beyond.The.End}, range} = resolve(project, code)
+      assert range.start.line == 1
+      assert range.start.character == 1
+      assert range.end.line == 1
+      assert range.end.character == 15
+    end
+
+    test "fails with an extra space following the module", %{project: project} do
+      code = ~q[
+        Beyond.The.End |
+      ]
+
+      assert {:error, :not_found} = resolve(project, code)
+    end
+
+    test "fails immediately preceeding the module", %{project: project} do
+      code = ~q[
+        | Before.The.Beginning
+      ]
+
+      assert {:error, :not_found} = resolve(project, code)
+    end
+
+    test "resolves module segments at and before the cursor", %{project: project} do
+      code = ~q[
+        In.|The.Middle
+      ]
+
+      assert {:ok, {:module, In.The}, range} = resolve(project, code)
+      assert range.start.line == 1
```

Comment:
To be honest, I prefer the range assert pattern as the same as the definition** used, rather than specific lines and characters. This allows us to see what we are resolving clearly.

assert resolved_range = "In.«The».Middle"

@scohen What do you think?

Reply:
Found the existing usage [here](https://github.com/lexical-lsp/lexical/blob/d9dcae41e03254f589a7c44e058013fe098043b4/apps/server/test/lexical/server/code_intelligence/definition_test.exs#L258) -- I’ll happily extract this and use it in these tests.